File: backend/app/simulation/shading_screen.py
"""Shade / thermal screen transmittance helpers."""

CLOSED_SCREEN_TRANSMITTANCE = 0.30


def effective_solar_transmittance(
    cover_transmittance: float,
    installed: bool,
    deployment_pct: float,
) -> float:
    """Combined glazing + retractable screen solar transmittance (0–1)."""
    if not installed or deployment_pct <= 0:
        return cover_transmittance

    deploy = max(0.0, min(100.0, deployment_pct)) / 100.0
    screen_factor = 1.0 - deploy * (1.0 - CLOSED_SCREEN_TRANSMITTANCE)
    return cover_transmittance * screen_factor

File: frontend/src/lib/shadingScreen.ts
import type { CoveringMaterial, ShadingScreen } from "@/types/greenhouse";

/** Solar transmittance of a fully closed aluminized thermal shade screen. */
export const CLOSED_SCREEN_TRANSMITTANCE = 0.3;

/** Combined cover + screen solar transmittance (0–1). */
export function effectiveSolarTransmittance(
  covering: Pick<CoveringMaterial, "transmittance">,
  screen: ShadingScreen,
): number {
  if (!screen.installed || screen.deploymentPct <= 0) {
    return covering.transmittance;
  }

  const deploy = Math.min(100, Math.max(0, screen.deploymentPct)) / 100;
  const screenFactor = 1 - deploy * (1 - CLOSED_SCREEN_TRANSMITTANCE);
  return covering.transmittance * screenFactor;
}
